src/schedule/zephyr_domain.c

```diff
@@ -159,6 +160,8 @@ static int zephyr_domain_register(struct ll_schedule_domain *domain,
 
 	k_thread_start(thread);
 
+	spin_lock_irq(&domain->lock, flags);
```

Comment:
can we call the zephyr spinlock here too.

Reply:
@lgirdwood same here - the argument is of a different type and that spinlock is a member of `struct ll_schedule_domain` which is also used with XTOS. We probably want a tree-wide spin-lock migration to Zephyr.

Reply:
we do, pls add that to your tasks for v2.1